MetaMask/metamask-extension
# 17649 - Hardware wallet signing state

## Explanation

Issue explained via story:

1. User initiates a signing request with an unlocked Ledger
2. Confirmation Modal pops up, and no warning is present because the Ledger is unlocked
3. Ledger locks while the Confirmation is open (e.g. the user may be hesitating the confirmation for a bit, or may be playing around with fees etc.)
4. User hits "Sign" and the signing request fails silently as the Ledger is now locked

Fixes: 
#17649 
#17630
#17606

See related:
#17937 
#17940

## Screenshots/Screencaps

### Before

Signing request was allowed to proceed with a locked device:
![image](https://user-images.githubusercontent.com/82925669/219506982-e74fb027-8f15-4b4d-9f2b-4df9b37880ea.png)

Resulting in a silent error and no result:
![image](https://user-images.githubusercontent.com/82925669/219507094-ec030291-7609-4fbc-888c-227d51040fa8.png)

### After

Signing requests now monitor HD wallet state and displays an appropriate error message when locked.

Personal Sign:
![image](https://user-images.githubusercontent.com/82925669/230237941-69efdb95-3bf3-4b7f-af72-03316d1f8fe7.png)

Sign Typed Data:
![image](https://user-images.githubusercontent.com/82925669/230238019-f1118926-ce5d-4c20-b12f-181194231784.png)

Sign Typed Data V3:
![image](https://user-images.githubusercontent.com/82925669/230238145-085abf80-b0fb-46d8-a792-078bb3ed8c51.png)

Sign Typed Data V4:
![image](https://user-images.githubusercontent.com/82925669/230238217-08f1692e-c8a6-4fd9-953f-95925aedfb63.png)

Sign in With Ethereum:
![image](https://user-images.githubusercontent.com/82925669/230238262-a7f833b6-907d-4335-beb3-9def157451bc.png)

Send Transaction:
![image](https://user-images.githubusercontent.com/82925669/230238351-51e1ec87-5ce6-4471-8ddd-9434770a1c15.png)

## Manual Testing Steps

1. Initiate signing request with Ledger
2. While on the confirm screen, wait for the Ledger to lock. Signing request should be blocked with ledgerLocked warning message showing
3. Unlock Ledger. The ledgerLocked warning should no longer be present and the signing request should now be able to proceed
4. Sign request and proceed as normal

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
